### PR TITLE
dont quit if node does not have subnet annotation

### PR DESCRIPTION
**Pull request by jluhrsen**
in some cases, multiple nodes will have to sync and we may not have enough subnets available for all of them. In that case we don't want to quit the sync on other nodes if one is found to not have the subnet annotation as the others may still have it.

JIRA: https://issues.redhat.com/browse/OCPBUGS-18598

**- What this PR does and why is it needed**

**- Special notes for reviewers**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: go-controller/pkg/node/default_node_network_controller.go [hunk 1]
```diff
@@ -872,7 +872,7 @@ func (nc *DefaultNodeNetworkController) Start(ctx context.Context) error {
 						nodeSubnets, err := util.ParseNodeHostSubnetAnnotation(&node, types.DefaultNetworkName)
 						if err != nil {
 							err1 = fmt.Errorf("unable to fetch node-subnet annotation for node %s: err, %v", node.Name, err)
-							return false, nil
+							continue
```
Comment (flavio-fernandes) [inline go-controller/pkg/node/default_node_network_controller.go:875]: hm... so, if it turns out that no nodes have a subnet annotation, we will skip them all and that is ok? In other words, what we really care here is that if the node has a subnet annotation, it better pass `checkOVNSBNodeLRSR`. Either that, or it is okay for it not to have that annotation. Right?

Comment (jcaamano) [inline go-controller/pkg/node/default_node_network_controller.go:875]: nit: I guess we don't need the err1 defined above for anything

Comment (trozet) [inline go-controller/pkg/node/default_node_network_controller.go:875]: @jluhrsen while you are here, coudl you also fix line 888, 908 : 
```
klog.Infof("Upgrade Hack: Syncing nodes took %v", time.Since(start))
```

this always reports time since the loop started, not the time since the these things actually completed. In other words if the node check passed in 5 seconds, and then after this services are checked and fail, then the entire loop will restart, and now the line will print time a new longer time for how long it took to sync nodes.

Comment (tssurya) [reply go-controller/pkg/node/default_node_network_controller.go:875]: right if the node is plumbed in, then we expect the route to exist for overlay traffic, if it doesn't have the node add done correctly because of lack of annotation, then we ignore that node

Comment (tssurya) [inline go-controller/pkg/node/default_node_network_controller.go:875]: shouldn't the continue be done only if this annotation is "notFound" ? other types of errors should they be continued as well?

Comment (tssurya) [reply go-controller/pkg/node/default_node_network_controller.go:875]: i see how this can be confusing :) i did this intentionally and would subtract the time printed from previous time print, but +1 to this comment to make this better.

Comment (tssurya) [reply go-controller/pkg/node/default_node_network_controller.go:875]: also should we print a message saying you skipped this node because you did not find the annotation?

Comment (jluhrsen) [reply go-controller/pkg/node/default_node_network_controller.go]: not sure my update is what you really wanted. the only possible error that we could get from ParseNodeHostSubnetAnnotations was that it was missing, but I *think* have it very specifically matching on the error message for that now. any new error that comes down the road wouldn't end up continuing.

Comment (jluhrsen) [reply go-controller/pkg/node/default_node_network_controller.go]: ok to just bump the `start := time.Now()` to the beginning of each PollUntilContextTimeout() ? that should reset it each time we work through the syncs

Comment (jluhrsen) [reply go-controller/pkg/node/default_node_network_controller.go]: @tssurya , with my latest update we are going to spam the log message until the node gets it's annotation. i know this is a corner case, but until the CIDR is expanded we will get a lot of them...

but, I did test these current changes locally and they seem to work as expected

File: go-controller/pkg/util/util.go [hunk 2]
```diff
@@ -180,22 +180,22 @@ func GetHybridOverlayPortName(nodeName string) string {
 	return "int-" + nodeName
 }
 
-type annotationNotSetError struct {
-	msg string
+type AnnotationNotSetError struct {
```
Comment (trozet) [inline go-controller/pkg/util/util.go:183]: you dont need to export all this, you can just call IsAnnotationNotSetError

File: go-controller/pkg/node/default_node_network_controller.go [hunk 3]
```diff
@@ -871,6 +871,12 @@ func (nc *DefaultNodeNetworkController) Start(ctx context.Context) error {
 					if nc.name != node.Name && util.GetNodeZone(&node) != config.Default.Zone && !houtil.IsHybridOverlayNode(&node) {
 						nodeSubnets, err := util.ParseNodeHostSubnetAnnotation(&node, types.DefaultNetworkName)
 						if err != nil {
+							if annotationError, ok := err.(*util.AnnotationNotSetError); ok {
+								if strings.Contains(annotationError.Msg, fmt.Sprintf("node %q has no k8s.ovn.org/node-subnets annotation for network %s", node.Name, types.DefaultNetworkName)) {
```
Comment (trozet) [inline go-controller/pkg/node/default_node_network_controller.go:875]: why do you need to string match here? you already know its an annotation not set, so this seems like overkill and not maintainable long term if the message changes.

**Review by trozet — CHANGES_REQUESTED** [review]

**Review by trozet — APPROVED** [review]

**Review by tssurya — COMMENTED** [review]
/lgtm
